modules/repository-azure/src/main/java/org/elasticsearch/repositories/azure/AzureBlobStore.java

```diff
@@ -278,48 +283,40 @@ private static void filterDeleteExceptionsAndRethrow(Exception e, IOException ex
-     * @see <a href="https://github.com/Azure/azure-storage-java/issues/538">Java SDK issue</a>
-     * @see <a href="https://github.com/elastic/elasticsearch/pull/65140#discussion_r528752070">Discussion on implementing PR</a>
-     */
     @Override
-    public void deleteBlobsIgnoringIfNotExists(OperationPurpose purpose, Iterator<String> blobs) {
-        if (blobs.hasNext() == false) {
+    public void deleteBlobsIgnoringIfNotExists(OperationPurpose purpose, Iterator<String> blobNames) throws IOException {
+        if (blobNames.hasNext() == false) {
             return;
         }
+        SocketAccess.doPrivilegedVoidException(() -> deleteListOfBlobs(getAzureBlobServiceClientClient(purpose), blobNames));
+    }
 
-        BlobServiceAsyncClient asyncClient = asyncClient(purpose);
-        SocketAccess.doPrivilegedVoidException(() -> {
-            final BlobContainerAsyncClient blobContainerClient = asyncClient.getBlobContainerAsyncClient(container);
+    private void deleteListOfBlobs(AzureBlobServiceClient azureBlobServiceClient, Iterator<String> blobNames) throws IOException {
+        // We need to use a container-scoped BlobBatchClient, so the restype=container parameter
+        // is sent, and we can support all SAS token types
+        // See https://learn.microsoft.com/en-us/rest/api/storageservices/blob-batch?tabs=shared-access-signatures#authorization
+        BlobBatchClient batchClient = new BlobBatchClientBuilder(
+            azureBlobServiceClient.getAsyncClient().getBlobContainerAsyncClient(container)
+        ).buildClient();
+        while (blobNames.hasNext()) {
+            final BlobBatch currentBatch = batchClient.getBlobBatch();
+            int counter = 0;
+            while (counter < MAX_ELEMENTS_PER_BATCH && blobNames.hasNext()) {
+                currentBatch.deleteBlob(container, blobNames.next());
+                counter++;
+            }
             try {
-                Flux.fromStream(StreamSupport.stream(Spliterators.spliteratorUnknownSize(blobs, Spliterator.ORDERED), false))
-                    .flatMap(blob -> getDeleteTask(blob, blobContainerClient.getBlobAsyncClient(blob)), CONCURRENT_DELETES)
-                    .then()
-                    .block();
-            } catch (Exception e) {
-                filterDeleteExceptionsAndRethrow(e, new IOException("Unable to delete blobs"));
+                batchClient.submitBatch(currentBatch);
+            } catch (BlobBatchStorageException bbse) {
+                final Iterable<BlobStorageException> batchExceptions = bbse.getBatchExceptions();
+                for (BlobStorageException bse : batchExceptions) {
+                    // If one of the requests failed with something other than a 404, throw the encompassing exception
+                    if (bse.getStatusCode() != RestStatus.NOT_FOUND.getStatus()) {
+                        throw new IOException("Failed to delete batch", bbse);
+                    }
+                }
```

Comment:
We should probably keep trying to delete the remaining blobs and just accumulate any exceptions to throw once we've exhausted the iterator. (I think this is how S3 works)